### PR TITLE
Ports: Move build directories into `Build/`

### PR DESCRIPTION
**Pull request by timschumi**
No more conflicts between architectures, and no more SourceForge bans. :^)

### CODE REVIEW

File: Toolchain/CMake/meson-cross-file-Clang.txt.in [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[binaries]
+c = '@SERENITY_ARCH@-pc-serenity-clang'
+cpp = '@SERENITY_ARCH@-pc-serenity-clang++'
+ar = '@SERENITY_ARCH@-pc-serenity-ar'
+ranlib = '@SERENITY_ARCH@-pc-serenity-ranlib'
```
[the remark this answers is not in the document]
Comment (timschumi) [reply Toolchain/CMake/meson-cross-file-Clang.txt.in:5]: I have to admit, I haven't gotten around to testing the new meson files with a clang build yet. The distinction so far exists only to get rid of the eyesore that is `gcc` in a Clang-tailored meson file, and on the off-chance that the rest of the settings just work (which they apparently won't).

**Comment by circl-lastname** [comment]
Regarding the Meson patches, why is it located in Toolchain/CMake? I actually do have practically the exact same patch that I didn't PR as I found the real issue with powdertoy. I put it in Toolchain/Meson, I also called the configured file MesonCrossFile.txt, as I thought that would fit in more with the existing filename conventions.

**Review by linusg — APPROVED** [review]
Very cool!

**Comment by circl-lastname** [comment]
Bruh

**Comment by circl-lastname** [comment]
What about the filename?

**Comment by timschumi** [comment]
CMake likes to use camel case, while most meson projects appear to use `-` or `_` for naming their cross files. While I don't have any hard feelings either way, the latter is why I chose to name the cross file as it is.

Regarding the `@SERENITY_ARCH@-pc-serenity-clang` vs. `--target` convention, we probably plan to move forward with the former due to better compatibility. However, we haven't forced a toolchain rebuild ever since we added those symlinks, so those only get created manually when compiling ports, which is why they are fine for the meson cross file, but not for the `CMakeToolchain.txt`.

**Pull request merged**